tools/cnfs/cnfs_gen.c

```diff
@@ -0,0 +1,157 @@
+#include <stdio.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <dirent.h>
+#include <stdint.h>
+
+int stringcmp( const void * a, const void * b );
+
+#define MAX_FILES 4096
+
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+struct fileEntry
+{
+    char * filename;
+    uint8_t * data;
+    int offset;
+    int len;
+} entries[MAX_FILES];
+int nr_file = 0;
+
+char * filelist[MAX_FILES];
+
+int stringcmp( const void * a, const void * b )
+{
+    return strcmp( *(const char* const *)a, *(const char* const *)b );
+}
+
+int main( int argc, char ** argv )
+{
+    if( argc != 4 )
+    {
+        fprintf( stderr, "Error: Usage: cnfs_gen folder/ image.c image.h\n" );
+        return -5;
+    }
+    char tfolder[PATH_MAX];
+    snprintf( tfolder, PATH_MAX, "%s", argv[1] );
+    struct dirent *dp;
+    DIR *dir = opendir(tfolder);
+    if( !dir )
+    {
+        fprintf( stderr, "Error: Can't open %s\n", argv[1] );
+        return -5;
+    }
+    int offset = 0;
+    int numfiles_in = 0;
+    while ((dp=readdir(dir))) // if dp is null, there's no more content to read
```

Comment:
This isn't recursive, and doesnt have to be. The preprocessor spits out a flat folder of files.

Reply:
SPIFFS isn't recursive anyway, so it wouldn't work even if it were.  But it would be trivial to make this system recursive without any code changes on the swadge-side.

Reply:
Not fixing (for now) since it doesn't make snese.

Reply:
And nothing broken, this was more of a note-to-self